client/src/App.js

```diff
@@ -0,0 +1,226 @@
+import "./App.css";
+import Axios from "axios";
+import { useNavigate } from "react-router-dom";
+import React, { useEffect, useState } from "react";
+
+function App() {
+  const getInitialState = () => {
+    const gender = "Male";
+    return gender;
+  };
+  const navigate = useNavigate();
+
+  const [name, setName] = useState("");
+  const [dob, setDob] = useState("");
+  const [dept, setDept] = useState("IT");
+  const [gender, setGender] = useState(getInitialState);
+
+  const [newName, setNewName] = useState("");
+  const [studentList, setStudentList] = useState([]);
+
+  const [deptList, setDeptList] = useState([]);
+
+  const handleGender = (e) => {
+    setGender(e.target.value);
+  };
+
+  const handleDepartment = (e) => {
+    setDept(e.target.value);
+  };
+
+  // ------------------------------------------------------
+  const [role, setRole] = useState("");
+  const [username, setUsername] = useState("");
+  Axios.defaults.withCredentials = true;
+
+  useEffect(() => {
+    Axios.get("http://localhost:3001/login").then((response) => {
+      if (response.data.loggedIn == true) {
+        setRole(response.data.user[0].role);
+        setUsername(response.data.user[0].username);
+        // console.log(response);
+      }
+    });
+  }, []);
+  // -------------------------------------------------------
+
+  // FUNCTIONS ---------------------------------------------------------------------------------------------------------------------------------------
+
+  const getDepartments = () => {
+    Axios.get("http://localhost:3001/departments").then((response) => {
+      setDeptList(response.data);
+    });
+  };
+
+  const addDepartment = () => {
+    Axios.post("http://localhost:3001/createDepartment", {
+      dept: dept,
+    }).then((response) => {
+      console.log("New Department Added!");
+    });
+  };
+
+  const getStudents = () => {
+    Axios.get("http://localhost:3001/students").then((response) => {
+      setStudentList(response.data);
+    });
+  };
+
+  const addStudent = () => {
+    Axios.post("http://localhost:3001/create", {
```

Comment:
Move api domain to config, so it will not be repeated in every URL, we will have single place to update if require in future.

Reply:
I am configuring routes at the moment. Will add this under "POST" request for student.